--- perl-5.14.0/uconfig.sh.orig	2010-03-14 22:15:34.000000000 +0100
+++ perl-5.14.0/uconfig.sh	2010-04-13 18:38:00.353231241 +0200
@@ -5,12 +5,7 @@
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.24/unknown'
-archlibexp='/usr/local/lib/perl5/5.24/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -44,7 +39,7 @@
 d_accessx='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_atolf='undef'
@@ -232,7 +232,7 @@
 d_link='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_longdbl='undef'
@@ -662,14 +657,10 @@
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.24'
-privlibexp='/usr/local/lib/perl5/5.24'
 procselfexe=''
 prototype='undef'
@@ -700,8 +691,6 @@
 sPRIXU64='"LX"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
--- p/uconfig.h~	2012-08-03 19:35:26.000000000 +0200
+++ p/uconfig.h	2012-08-28 19:39:39.192490428 +0200
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * ce61c4f18e82f6356cdae44c5336ceac1d3253033d7449651927f5633be5c3c2 config_h.SH
- * fc611849cb5b1e14ec1687b255dac15414cc5e2e11b192d94e08136cfe277f75 uconfig.sh
+ * 8e70a60e2dc426f9c04ffc90fa7269978ba975771590ada658b5f3eb8e516d3d uconfig.sh
  * ex: set ro: */
