--- perl-5.10.1/utils/h2ph.PL.orig	2009-08-25 10:14:50.000000000 +0200
+++ perl-5.10.1/utils/h2ph.PL	2009-08-25 10:23:00.000000000 +0200
@@ -722,6 +722,7 @@
 
     return if ($file eq "-");
 
+    (my $filedir = $file) =~ s/[^\/]*$//;
     open HEADER, $file or return;
         while (defined($line = <HEADER>)) {
             while (/\\$/) { # Handle continuation lines
@@ -736,7 +737,9 @@
                     $new_file = "$1/$new_file";
                 }
                 push(@ARGV, $new_file) unless $Is_converted{$new_file};
-            }
+            } elsif ($line =~ /^#\s*include\s+\"(.*?)\"/) {
+		    push(@ARGV, $filedir . $new_file) unless $Is_converted{$filedir . $new_file};
+	    }
         }
     close HEADER;
 }
