--- perl-5.14.0/uconfig.sh.orig	2010-03-14 22:15:34.000000000 +0100
+++ perl-5.14.0/uconfig.sh	2010-04-13 18:38:00.35323141 +0200
@@ -5,12 +5,7 @@
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.30/unknown'
-archlibexp='/usr/local/lib/perl5/5.30/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -44,7 +39,7 @@
 d_accessx='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_atolf='undef'
@@ -232,7 +232,7 @@
 d_link='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_longdbl='undef'
@@ -662,14 +657,10 @@
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.30'
-privlibexp='/usr/local/lib/perl5/5.30'
 procselfexe=''
 prototype='undef'
@@ -700,8 +691,6 @@
 sPRIXU64='"LX"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
--- p/uconfig.h~	2012-08-03 19:35:26.000000000 +0200
+++ p/uconfig.h	2012-08-28 19:39:39.192490428 +0200
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * 6608de918c3c876975f74b684da2536ab1ee23459783d691ae02ce2526a497a7 config_h.SH
- * 2aaf18b9277e180fc5e5d60290ecb0c91fcac3531bd8825e5687a212daa586e9 uconfig.sh
+ * 700be6acc013e2e38111b8fd0c3029d9ca67436e6bb85c457050243683e30b92 uconfig.sh
  * ex: set ro: */
