--- pod/perluniintro.pod.orig	Thu Feb  6 12:45:57 2003
+++ pod/perluniintro.pod	Thu Feb  6 12:46:10 2003
@@ -363,7 +363,7 @@
 
 With the C<open> pragma you can use the C<:locale> layer
 
-    $ENV{LC_ALL} = $ENV{LANG} = 'ru_RU.KOI8-R';
+    BEGIN { $ENV{LC_ALL} = $ENV{LANG} = 'ru_RU.KOI8-R'; }
     # the :locale will probe the locale environment variables like LC_ALL
     use open OUT => ':locale'; # russki parusski
     open(O, ">koi8");
