diff -urN perl-5.6.1.orig/lib/ExtUtils/MakeMaker.pm perl-5.6.1/lib/ExtUtils/MakeMaker.pm
--- perl-5.6.1.orig/lib/ExtUtils/MakeMaker.pm	Fri Feb 23 03:57:55 2001
+++ perl-5.6.1/lib/ExtUtils/MakeMaker.pm	Thu Apr 12 20:43:14 2001
@@ -202,6 +202,7 @@
     PERL_ARCHLIB PERL_LIB PERL_SRC PERM_RW PERM_RWX
     PL_FILES PM PM_FILTER PMLIBDIRS POLLUTE PPM_INSTALL_EXEC
 	PPM_INSTALL_SCRIPT PREFIX
+    PRINT_PREREQ
     PREREQ_PM SKIP TYPEMAPS VERSION VERSION_FROM XS XSOPT XSPROTOARG
     XS_VERSION clean depend dist dynamic_lib linkext macro realclean
     tool_autosplit
@@ -319,6 +320,14 @@
 sub ExtUtils::MakeMaker::new {
     my($class,$self) = @_;
     my($key);
+
+    if ("@ARGV" =~ "PRINT_PREREQ") { 
+       foreach $prereq (sort keys %{$self->{PREREQ_PM}}) {
+           print "perl($prereq)>=$self->{PREREQ_PM}->{$prereq} ";
+       }
+       print "\n";
+       exit 0;
+    }
 
     print STDOUT "MakeMaker (v$VERSION)\n" if $Verbose;
     if (-f "MANIFEST" && ! -f "Makefile"){
