--- lib/ExtUtils/MM_Unix.pm.orig	Tue Feb 25 16:03:41 2003
+++ lib/ExtUtils/MM_Unix.pm	Tue Feb 25 16:19:23 2003
@@ -499,7 +499,7 @@
         # pathnames can have sharp signs in them; escape them so
         # make doesn't think it is a comment-start character.
         $self->{$tmp} =~ s/#/\\#/g;
-	push @m, "$tmp = $self->{$tmp}\n";
+	push @m, "$tmp = " . ($tmp =~ /^INSTALL(?!DIRS\b)/ ? '$(DESTDIR)' : '') . "$self->{$tmp}\n";
     }
 
     push @m, qq{
@@ -549,7 +549,7 @@
 	      /) 
     {
 	next unless defined $self->{$tmp};
-	push @m, "$tmp = $self->{$tmp}\n";
+	push @m, "$tmp = " . ($tmp =~ /^INSTALL/ ? '$(DESTDIR)' : '') . "$self->{$tmp}\n";
     }
 
     for $tmp (qw(
@@ -1884,9 +1884,9 @@
 
     if( $Config{usevendorprefix} ) {
         $Config_Override{installvendorman1dir} =
-          File::Spec->catdir($Config{vendorprefixexp}, 'man', 'man$(MAN1EXT)');
+          File::Spec->catdir($Config{vendorprefixexp}, 'share', 'man', 'man1');
         $Config_Override{installvendorman3dir} =
-          File::Spec->catdir($Config{vendorprefixexp}, 'man', 'man$(MAN3EXT)');
+          File::Spec->catdir($Config{vendorprefixexp}, 'share', 'man', 'manp');
     }
     else {
         $Config_Override{installvendorman1dir} = '';
@@ -1905,12 +1905,12 @@
     # it up.
     unless( $Config{installsiteman1dir} ) {
         $Config_Override{installsiteman1dir} = 
-          File::Spec->catdir($sprefix, 'man', 'man$(MAN1EXT)');
+          File::Spec->catdir($sprefix, 'share', 'man', 'man1');
     }
 
     unless( $Config{installsiteman3dir} ) {
         $Config_Override{installsiteman3dir} = 
-          File::Spec->catdir($sprefix, 'man', 'man$(MAN3EXT)');
+          File::Spec->catdir($sprefix, 'share', 'man', 'manp');
     }
 
     unless( $Config{installsitebin} ) {
