--- perl-5.6.1/Configure.orig	Mon Mar 19 04:03:33 2001
+++ perl-5.6.1/Configure	Tue Dec 24 20:12:56 2002
@@ -1030,10 +1030,9 @@
 
 : List of libraries we want.
 : If anyone needs -lnet, put it in a hint file.
-libswanted='sfio socket bind inet nsl nm ndbm gdbm dbm db malloc dl'
-libswanted="$libswanted dld ld sun m c cposix posix"
-libswanted="$libswanted ndir dir crypt sec"
-libswanted="$libswanted ucb bsd BSD PW x iconv util"
+libswanted='ndbm gdbm db dl'
+libswanted="$libswanted m c"
+libswanted="$libswanted crypt"
 : We probably want to search /usr/shlib before most other libraries.
 : This is only used by the lib/ExtUtils/MakeMaker.pm routine extliblist.
 glibpth=`echo " $glibpth " | sed -e 's! /usr/shlib ! !'`
