diff -urNp -x '*.orig' perl-5.32.1.org/uconfig.h perl-5.32.1/uconfig.h
--- perl-5.32.1.org/uconfig.h	2020-12-18 11:04:36.000000000 +0100
+++ perl-5.32.1/uconfig.h	2021-04-01 22:48:37.569619974 +0200
@@ -5261,5 +5261,5 @@
 
 /* Generated from:
  * 87e5998978daf803d19866c43bca24d7c01dc74119650db16f8d18d83f355da9 config_h.SH
- * 1a5fe19cbcfd68ba70230580fd344189b5c78c11b2285efd5976366e51b3257e uconfig.sh
+ * 6422951512b8b4f51af01c1445068645c09438db5b0e56f7e711fd9accb4942f uconfig.sh
  * ex: set ro: */
diff -urNp -x '*.orig' perl-5.32.1.org/uconfig.sh perl-5.32.1/uconfig.sh
--- perl-5.32.1.org/uconfig.sh	2020-12-18 11:04:36.000000000 +0100
+++ perl-5.32.1/uconfig.sh	2021-04-01 22:48:37.569619974 +0200
@@ -6,12 +6,7 @@ afs='false'
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.36/unknown'
-archlibexp='/usr/local/lib/perl5/5.36/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -48,7 +43,7 @@ d_accessx='undef'
 d_acosh='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_asinh='undef'
@@ -292,7 +287,7 @@ d_llroundl='undef'
 d_localeconv_l='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_log1p='undef'
@@ -774,16 +769,12 @@ obj_ext='.o'
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 perl_thread_local=''
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.36'
-privlibexp='/usr/local/lib/perl5/5.36'
 procselfexe=''
 prototype='undef'
 ptrsize='4'
@@ -813,8 +804,6 @@ sPRIu64='"Lu"'
 sPRIx64='"Lx"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
