--- perl-5.6.0/installperl.pius	Wed Mar  8 13:22:41 2000
+++ perl-5.6.0/installperl	Sun Jul  2 13:36:20 2000
@@ -143,8 +143,6 @@
 
 # Do some quick sanity checks.
 
-if ($d_dosuid && $>) { die "You must run as root to install suidperl\n"; }
-
    $installbin		|| die "No installbin directory in config.sh\n";
 -d $installbin		|| mkpath($installbin, 1, 0777);
 -d $installbin		|| $nonono || die "$installbin is not a directory\n";
@@ -224,7 +222,7 @@
 safe_unlink("$installbin/s$perl_verbase$ver$exe_ext");
 if ($d_dosuid) {
     copy("suidperl$exe_ext", "$installbin/s$perl_verbase$ver$exe_ext");
-    chmod(04711, "$installbin/s$perl_verbase$ver$exe_ext");
+    chmod(0755, "$installbin/s$perl_verbase$ver$exe_ext");
 }
 
 # Install library files.
