--- perl-5.14.0/uconfig.sh.orig	2010-03-14 22:15:34.000000000 +0100
+++ perl-5.14.0/uconfig.sh	2010-04-13 18:38:00.353231241 +0200
@@ -5,12 +5,7 @@
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.22/unknown'
-archlibexp='/usr/local/lib/perl5/5.22/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -44,7 +39,7 @@
 d_accessx='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_atolf='undef'
@@ -232,7 +232,7 @@
 d_link='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_longdbl='undef'
@@ -662,14 +657,10 @@
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.22'
-privlibexp='/usr/local/lib/perl5/5.22'
 procselfexe=''
 prototype='undef'
@@ -700,8 +691,6 @@
 sPRIXU64='"LX"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
--- p/uconfig.h~	2012-08-03 19:35:26.000000000 +0200
+++ p/uconfig.h	2012-08-28 19:39:39.192490428 +0200
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * 496e563499c7b715275d61ae663d25dd20d963c75f9d3ee7850dae949df14136 config_h.SH
- * 45fb46f69ab3d7fd8ea14fc29d490d044cbdd81e8a0111bc0b37aff4321d6182 uconfig.sh
+ * 015e6df6c8df54d93508b9c0532aadd91f1614597edc82ac2d9cbecc5732654b uconfig.sh
  * ex: set ro: */
