--- perl-5.14.0/uconfig.sh.orig	2010-03-14 22:15:34.000000000 +0100
+++ perl-5.14.0/uconfig.sh	2010-04-13 18:38:00.35323141 +0200
@@ -5,12 +5,7 @@
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.26/unknown'
-archlibexp='/usr/local/lib/perl5/5.26/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -44,7 +39,7 @@
 d_accessx='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_atolf='undef'
@@ -232,7 +232,7 @@
 d_link='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_longdbl='undef'
@@ -662,14 +657,10 @@
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.26'
-privlibexp='/usr/local/lib/perl5/5.26'
 procselfexe=''
 prototype='undef'
@@ -700,8 +691,6 @@
 sPRIXU64='"LX"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
--- p/uconfig.h~	2012-08-03 19:35:26.000000000 +0200
+++ p/uconfig.h	2012-08-28 19:39:39.192490428 +0200
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * a0cea6273c16c1c5d8625665c84bda01926ce960c26873d4e5596b5b02a53e92 config_h.SH
- * b1c042726e6a17692921b6947c4e7c196c5c0cee011667ba719b15af7c31e669 uconfig.sh
+ * e52958b2fc077af6dfab034b997d83d2697d5393063c23449e16d225b491364a uconfig.sh
  * ex: set ro: */
