--- perl-5.6.0/ext/Sys/Syslog/Syslog.pm.chmou	Tue Mar 14 17:29:25 2000
+++ perl-5.6.0/ext/Sys/Syslog/Syslog.pm	Sat Dec 16 13:46:10 2000
@@ -274,7 +274,7 @@
         socket(SYSLOG,AF_INET,SOCK_DGRAM,$udp)           || croak "socket: $!";
         connect(SYSLOG,$that)                            || croak "connect: $!";
     } else {
-        my $syslog = &_PATH_LOG                          || croak "_PATH_LOG not found in syslog.ph";
+        my $syslog = &_PATH_LOG()                          || croak "_PATH_LOG not found in syslog.ph";
         my $that = sockaddr_un($syslog)                  || croak "Can't locate $syslog";
         socket(SYSLOG,AF_UNIX,SOCK_STREAM,0)             || croak "socket: $!";
         if (!connect(SYSLOG,$that)) {
