diff -urNp -x '*.orig' perl-5.32.1.org/ext/Errno/Errno_pm.PL perl-5.32.1/ext/Errno/Errno_pm.PL
--- perl-5.32.1.org/ext/Errno/Errno_pm.PL	2020-12-18 11:04:36.000000000 +0100
+++ perl-5.32.1/ext/Errno/Errno_pm.PL	2021-04-01 22:48:31.609573602 +0200
@@ -67,6 +67,13 @@ sub process_file {
             warn "Cannot open '$file'";
             return;
 	}     
+    } elsif ($^O eq 'linux') {
+	# With the -dM option option, gcc outputs every #define's it finds
+	my $cpp = default_cpp();
+	unless(open(FH, "$cpp -dM < $file |")) {
+            warn "Cannot open '$file'";
+            return;
+	}
     } else {
 	unless(open(FH, '<', $file)) {
 	    # This file could be a temporary file created by cppstdin
