diff -urN perl-5.6.1/find-perl-provides perl-5.6.1-grabba/find-perl-provides
--- perl-5.6.1/find-perl-provides	Thu Jan  1 00:00:00 1970
+++ perl-5.6.1-grabba/find-perl-provides	Sun Mar  3 14:41:39 2002
@@ -0,0 +1,23 @@
+#!/bin/sh
+ulimit -c 0
+
+export LD_LIBRARY_PATH="FPPATH"
+
+filelist=`sed "s/['\"]/\\\&/g"`
+if [ -f __rpm_noautoprovfiles ] ; then
+	for i in `cat __rpm_noautoprovfiles`; do
+		filelist=`echo $filelist | sed "s![[:space:]]*$i[[:space:]]*!!g"`
+	done
+fi
+
+
+echo $filelist|/usr/lib/rpm/find-provides
+provides_perl=`FPPATH/find-perl.prov $filelist`
+if [ -f __rpm_noautoprov ] ; then
+	for i in `cat __rpm_noautoprov`; do
+		provides_perl=`echo $provides_perl | sed "s!\<$i[[:space:]]*!!g"`
+	done
+fi
+
+echo "$provides_perl"
+
