--- perl-5.6.0/lib/ExtUtils/Liblist.pm.orig	Wed Mar 22 17:16:31 2000
+++ perl-5.6.0/lib/ExtUtils/Liblist.pm	Fri Aug 18 01:03:48 2000
@@ -39,6 +39,7 @@
     my(@searchpath); # from "-L/path" entries in $potential_libs
     my(@libpath) = split " ", $Config{'libpth'};
     my(@ldloadlibs, @bsloadlibs, @extralibs, @ld_run_path, %ld_run_path_seen);
+    $ld_run_path_seen{$_} = 1 foreach qw(/usr/X11R6/lib /usr/lib /lib);
     my($fullname, $thislib, $thispth, @fullname);
     my($pwd) = cwd(); # from Cwd.pm
     my($found) = 0;
--- perl-5.6.0/lib/ExtUtils/MM_Unix.pm~	Fri Aug 18 01:03:48 2000
+++ perl-5.6.0/lib/ExtUtils/MM_Unix.pm	Fri Aug 18 01:04:22 2000
@@ -529,9 +529,16 @@
 };
     my($tmp);
     for $tmp (qw/
-	 EXTRALIBS LDLOADLIBS BSLOADLIBS LD_RUN_PATH
+	 EXTRALIBS LDLOADLIBS BSLOADLIBS
 	 /) {
 	next unless defined $self->{$tmp};
+	push @m, "$tmp = $self->{$tmp}\n";
+    }
+    # don't set LD_RUN_PATH if empty
+    for $tmp (qw/
+	 LD_RUN_PATH
+	 /) {
+	next unless $self->{$tmp};
 	push @m, "$tmp = $self->{$tmp}\n";
     }
     return join "", @m;
