diff -urN perl5.005_03.old/lib/File/Spec/Functions.pm perl5.005_03/lib/File/Spec/Functions.pm
--- perl5.005_03.old/lib/File/Spec/Functions.pm	Thu Jan  1 01:00:00 1970
+++ perl5.005_03/lib/File/Spec/Functions.pm	Sun Dec  6 05:22:30 1998
@@ -0,0 +1,37 @@
+package File::Spec::Functions;
+
+use File::Spec;
+use strict;
+
+use vars qw(@ISA @EXPORT);
+
+require Exporter;
+
+@ISA = qw(Exporter);
+
+@EXPORT = qw(
+	canonpath
+	catdir
+	catfile
+	curdir
+	rootdir
+	updir
+	no_upwards
+	file_name_is_absolute
+	path
+	nativename
+);
+
+sub canonpath { File::Spec->canonpath(@_); }
+sub catdir { File::Spec->catdir(@_); }
+sub catfile { File::Spec->catfile(@_); }
+sub curdir { File::Spec->curdir(@_); }
+sub rootdir { File::Spec->rootdir(@_); }
+sub updir { File::Spec->updir(@_); }
+sub no_upwards { File::Spec->no_upwards(@_); }
+sub file_name_is_absolute { File::Spec->file_name_is_absolute(@_); }
+sub path { File::Spec->path(@_); }
+sub nativename { File::Spec->nativename(@_); }
+
+1;
+
diff -urN perl5.005_03.old/lib/File/Spec/Mac.pm perl5.005_03/lib/File/Spec/Mac.pm
--- perl5.005_03.old/lib/File/Spec/Mac.pm	Thu Jan  7 05:41:53 1999
+++ perl5.005_03/lib/File/Spec/Mac.pm	Sun Feb 22 16:31:58 1998
@@ -52,7 +52,7 @@
 	  File::Spec->catdir(split(":",$path)) eq $path
 
 But because of the nature of Macintosh paths, some additional 
-possibilities are allowed to make using this routine give reasonable results 
+possibilities are allowed to make using this routine give resonable results 
 for some common situations.  Here are the rules that are used.  Each 
 argument has its trailing ":" removed.  Each argument, except the first,
 has its leading ":" removed.  They are then joined together by a ":".
@@ -78,7 +78,7 @@
 	  File::Spec->catfile("LWP","Protocol","http.pm")
 
 be relative or absolute?  There's no way of telling except by checking for the
-existence of LWP: or :LWP, and even there he may mean a dismounted volume or
+existance of LWP: or :LWP, and even there he may mean a dismounted volume or
 a relative path in a different directory (like in @INC).   So those checks
 aren't done here. This routine will treat this as absolute.
 
diff -urN perl5.005_03.old/lib/File/Spec.pm perl5.005_03/lib/File/Spec.pm
--- perl5.005_03.old/lib/File/Spec.pm	Thu Jan  7 05:41:53 1999
+++ perl5.005_03/lib/File/Spec.pm	Sun Dec  6 05:48:23 1998
@@ -14,7 +14,7 @@
 use strict;
 use vars qw(@ISA $VERSION $Verbose);
 
-$VERSION = '0.6';
+$VERSION = '0.7';
 
 $Verbose = 0;
 
@@ -52,11 +52,15 @@
 
 =head1 SYNOPSIS
 
-C<use File::Spec;>
+	use File::Spec;
 
-C<$x=File::Spec-E<gt>catfile('a','b','c');>
+	$x=File::Spec->catfile('a', 'b', 'c');
 
-which returns 'a/b/c' under Unix.
+which returns 'a/b/c' under Unix. Or:
+
+	use File::Spec::Functions;
+	
+	$x = catfile('a', 'b', 'c');
 
 =head1 DESCRIPTION
 
@@ -78,8 +82,8 @@
 	File::Spec::VMS
 
 The module appropriate for the current OS is automatically loaded by
-File::Spec. Since some modules (like VMS) make use of OS specific
-facilities, it may not be possible to load all modules under all operating
+File::Spec. Since some modules (like VMS) make use of facilities only under
+that OS, it may not be possible to load all modules under all operating
 systems.
 
 Since File::Spec is object oriented, subroutines should not called directly,
@@ -91,9 +95,17 @@
 
 	File::Spec->catfile('a','b');
 
-For a reference of available functions, please consult L<File::Spec::Unix>,
-which contains the entire set, and inherited by the modules for other
-platforms. For further information, please see L<File::Spec::Mac>,
+Alternatively, File::Spec::Functions can be used instead, which exports
+functions for all of the class methods provided by File::Spec, including
+catfile, so you may say:
+
+	use File::Spec::Functions;
+	
+	catfile('a','b');
+
+For a reference of available functions, pleaes consult L<File::Spec::Unix>,
+which contains the entire set, and which is inherited by the modules for
+other platforms. For further information, please see L<File::Spec::Mac>,
 L<File::Spec::OS2>, L<File::Spec::Win32>, or L<File::Spec::VMS>.
 
 =head1 SEE ALSO
