--- perl-5.6.1/Makefile.SH~	Sun Apr 15 18:20:15 2001
+++ perl-5.6.1/Makefile.SH	Sun Apr 15 18:36:49 2001
@@ -429,7 +429,8 @@
 	case "$useshrplib" in
 	true)
-		$spitshell >>Makefile <<'!NO!SUBS!'
-	$(LD) -o $@ $(SHRPLDFLAGS) perl$(OBJ_EXT) $(obj)
-!NO!SUBS!
+		$spitshell >>Makefile <<!GROK!THIS!
+	\$(LD) -o \$(LIBPERL).$version \$(SHRPLDFLAGS) -Wl,-soname,\$(LIBPERL).$version perl\$(OBJ_EXT) \$(obj)
+	ln -sf \$(LIBPERL).$version \$@
+!GROK!THIS!
 		case "$osname" in
 		aix)
