diff -urN perl-5.6.1.orig/lib/ExtUtils/Liblist.pm perl-5.6.1/lib/ExtUtils/Liblist.pm
--- perl-5.6.1.orig/lib/ExtUtils/Liblist.pm	Fri Feb 23 03:57:55 2001
+++ perl-5.6.1/lib/ExtUtils/Liblist.pm	Thu Apr 12 20:57:10 2001
@@ -60,6 +60,7 @@
     my(@searchpath); # from "-L/path" entries in $potential_libs
     my(@libpath) = split " ", $Config{'libpth'};
     my(@ldloadlibs, @bsloadlibs, @extralibs, @ld_run_path, %ld_run_path_seen);
+    $ld_run_path_seen{$_} = 1 foreach qw(/usr/X11R6/lib /usr/lib /lib);
     my(@libs, %libs_seen);
     my($fullname, $thislib, $thispth, @fullname);
     my($pwd) = cwd(); # from Cwd.pm
diff -urN perl-5.6.1.orig/lib/ExtUtils/MM_Unix.pm perl-5.6.1/lib/ExtUtils/MM_Unix.pm
--- perl-5.6.1.orig/lib/ExtUtils/MM_Unix.pm	Thu Apr 12 20:29:35 2001
+++ perl-5.6.1/lib/ExtUtils/MM_Unix.pm	Thu Apr 12 20:56:30 2001
@@ -524,9 +524,16 @@
 };
     my($tmp);
     for $tmp (qw/
-	 EXTRALIBS LDLOADLIBS BSLOADLIBS LD_RUN_PATH
+	 EXTRALIBS LDLOADLIBS BSLOADLIBS
 	 /) {
 	next unless defined $self->{$tmp};
+	push @m, "$tmp = $self->{$tmp}\n";
+    }
+    # don't set LD_RUN_PATH if empty
+    for $tmp (qw/
+	 LD_RUN_PATH
+	 /) {
+	next unless $self->{$tmp};
 	push @m, "$tmp = $self->{$tmp}\n";
     }
     return join "", @m;
