diff -ur perl5.005_61.old/lib/ExtUtils/MM_Unix.pm perl5.005_61/lib/ExtUtils/MM_Unix.pm
--- perl5.005_61.old/lib/ExtUtils/MM_Unix.pm	Fri Aug 20 17:51:31 1999
+++ perl5.005_61/lib/ExtUtils/MM_Unix.pm	Thu Aug 26 00:22:31 1999
@@ -527,8 +527,7 @@
 	      AR_STATIC_ARGS NAME DISTNAME NAME_SYM VERSION
 	      VERSION_SYM XS_VERSION INST_BIN INST_EXE INST_LIB
 	      INST_ARCHLIB INST_SCRIPT PREFIX  INSTALLDIRS
-	      INSTALLPRIVLIB INSTALLARCHLIB INSTALLSITELIB
-	      INSTALLSITEARCH INSTALLBIN INSTALLSCRIPT PERL_LIB
+	      PERL_LIB
 	      PERL_ARCHLIB SITELIBEXP SITEARCHEXP LIBPERL_A MYEXTLIB
 	      FIRST_MAKEFILE MAKE_APERL_FILE PERLMAINCC PERL_SRC
 	      PERL_INC PERL FULLPERL
@@ -538,6 +537,15 @@
 	push @m, "$tmp = $self->{$tmp}\n";
     }
 
+    for $tmp (qw/
+
+	      INSTALLPRIVLIB INSTALLARCHLIB INSTALLSITELIB
+	      INSTALLSITEARCH INSTALLBIN INSTALLSCRIPT 
+
+	      / ) {
+	next unless defined $self->{$tmp};
+	push @m, "$tmp = \$\(DESTDIR\)$self->{$tmp}\n";
+    }
     push @m, qq{
 VERSION_MACRO = VERSION
 DEFINE_VERSION = -D\$(VERSION_MACRO)=\\\"\$(VERSION)\\\"
@@ -580,17 +588,27 @@
 ";
 
     for $tmp (qw/
-	      INST_HTMLPRIVLIBDIR INSTALLHTMLPRIVLIBDIR
-	      INST_HTMLSITELIBDIR INSTALLHTMLSITELIBDIR
-	      INST_HTMLSCRIPTDIR  INSTALLHTMLSCRIPTDIR
-	      INST_HTMLLIBDIR                    HTMLEXT
-	      INST_MAN1DIR        INSTALLMAN1DIR MAN1EXT
-	      INST_MAN3DIR        INSTALLMAN3DIR MAN3EXT
+	      INST_HTMLPRIVLIBDIR 
+	      INST_HTMLSITELIBDIR 
+	      INST_HTMLSCRIPTDIR  
+	      INST_HTMLLIBDIR     HTMLEXT
+	      INST_MAN1DIR        MAN1EXT
+	      INST_MAN3DIR        MAN3EXT
 	      /) {
 	next unless defined $self->{$tmp};
 	push @m, "$tmp = $self->{$tmp}\n";
     }
 
+    for $tmp (qw/
+	      INSTALLHTMLPRIVLIBDIR
+	      INSTALLHTMLSITELIBDIR
+	      INSTALLHTMLSCRIPTDIR
+	      INSTALLMAN1DIR 
+	      INSTALLMAN3DIR 
+	      /) {
+	next unless defined $self->{$tmp};
+	push @m, "$tmp = \$\(DESTDIR\)$self->{$tmp}\n";
+    }
     for $tmp (qw(
 		PERM_RW PERM_RWX
 		)
