--- perl-5.10.1/installperl.orig	2009-08-14 00:40:10.000000000 +0200
+++ perl-5.10.1/installperl	2009-09-12 09:56:30.724923937 +0200
@@ -235,7 +235,7 @@
 
 # Do some quick sanity checks.
 
-if (!$opts{notify} && $d_dosuid && $>) { die "You must run as root to install suidperl\n"; }
+#if (!$opts{notify} && $d_dosuid && $>) { die "You must run as root to install suidperl\n"; }
 
    $installbin		|| die "No installbin directory in config.sh\n";
 -d $installbin		|| mkpath($installbin, $opts{verbose}, 0777);
@@ -344,7 +344,7 @@
 safe_unlink("$installbin/s$perl_verbase$ver$exe_ext");
 if ($d_dosuid) {
     copy("suidperl$exe_ext", "$installbin/s$perl_verbase$ver$exe_ext");
-    chmod(04711, "$installbin/s$perl_verbase$ver$exe_ext");
+    chmod(0755, "$installbin/s$perl_verbase$ver$exe_ext");
 }
 
 # Install library files.
