--- perl-5.14.0/uconfig.sh.orig	2010-03-14 22:15:34.000000000 +0100
+++ perl-5.14.0/uconfig.sh	2010-04-13 18:38:00.353231241 +0200
@@ -5,12 +5,7 @@
 afsroot='/afs'
 alignbytes='4'
 aphostname='/bin/hostname'
-archlib='/usr/local/lib/perl5/5.20/unknown'
-archlibexp='/usr/local/lib/perl5/5.20/unknown'
-archname='unknown'
 asctime_r_proto='0'
-bin='/usr/local/bin'
-binexp='/usr/local/bin'
 bootstrap_charset='undef'
 byteorder='1234'
 castflags='0'
@@ -44,7 +39,7 @@
 d_accessx='undef'
 d_aintl='undef'
 d_alarm='undef'
-d_archlib='undef'
+d_archlib='define'
 d_asctime64='undef'
 d_asctime_r='undef'
 d_atolf='undef'
@@ -232,7 +232,7 @@
 d_link='undef'
 d_localtime64='undef'
 d_localtime_r='undef'
-d_localtime_r_needs_tzset='undef'
+d_localtime_r_needs_tzset='define'
 d_locconv='undef'
 d_lockf='undef'
 d_longdbl='undef'
@@ -662,14 +657,10 @@
 old_pthread_create_joinable=''
 optimize=''
 orderlib='false'
-osname='unknown'
-osvers='unknown'
 otherlibdirs=' '
 package='perl5'
 perl_static_inline='static'
 phostname='hostname'
 pidtype=int
-privlib='/usr/local/lib/perl5/5.20'
-privlibexp='/usr/local/lib/perl5/5.20'
 procselfexe=''
 prototype='undef'
@@ -700,8 +691,6 @@
 sPRIXU64='"LX"'
 sSCNfldbl='"llf"'
 sched_yield='sched_yield()'
-scriptdir='/usr/local/bin'
-scriptdirexp='/usr/local/bin'
 seedfunc='Perl_drand48_init'
 selectminbits='32'
 selecttype=int
--- p/uconfig.h~	2012-08-03 19:35:26.000000000 +0200
+++ p/uconfig.h	2012-08-28 19:39:39.192490428 +0200
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * 7557e985de18f71e80f627226b454bc8eaf20477dcf0c45b5b2c51ec792f5c89 config_h.SH
- * dbc8d38ba52ae23e5423418bb3f56b1b6fcdaa82cf71ba0be3463e8221bfe0c0 uconfig.sh
+ * 47373608dbea89e81d56c421bde0da0969b5d7183ddff09ec5f65ab952ddc6cc uconfig.sh
  * ex: set ro: */
